reconcile/utils/jsonpath.py

```diff
@@ -1,11 +1,37 @@
-from functools import reduce
+import logging
+from functools import (
+    lru_cache,
+    reduce,
+)
 from itertools import zip_longest
 from typing import Optional
 
 import jsonpath_ng
 import jsonpath_ng.ext.filter
 
 
+@lru_cache(maxsize=256)
```

Comment:
> LGTM! Out of curiosity, what do you think the current performance impact is on MR checks?

this saves ~10 seconds since we parse a lot of expressions. it will not reduce the current MR runtimes because this usually runs in parallel with other integrations that take longer.

this is basically a preparation for the change-admission feature @janboll  and i plan to implement. change-admission will be an integration that runs before all other integrations and fails the PR check if a change is not allowed to come from a certain author (e.g. hack script updates from a non AppSRE team member). since change-admission will not run in parallel but before other integrations, it is crucial that it will run fast. this change is a small contribution to that.

Reply:
> Out of curiosity, how did you get to that number?

i had a look at the number of expressions we manage right now, added a buffer and then rounded up to a nice clean number :D